flycheck.el

```diff
@@ -4214,6 +4214,20 @@ universal prefix arg, and only the id with normal prefix arg."
     (`(eval ,_) t)
     (_ nil)))
 
+(defun flycheck-cwd-wrapper (checker)
+  "Function returning apropriate current working directory.
+
+Work out the working directory from which the CHECKER command
+will be called.  The default is the `default-directory' of the source-buffer
+from which flycheck is calling the checker.
+:cwd can be a list that contains a fixed path string or a function that will
+return a path.  The function needs to return an absolut path not a relative path."
+  (let ((cwd (flycheck-checker-get checker 'cwd)))
+    (or (and (functionp cwd) (funcall cwd))
+        (and (listp cwd) (stringp (car cwd)) (file-truename (car cwd)))
```

Comment:
As in for the use case. I'm setting my working directory to a constant. I'm setting it to '~/temp' since my syntax checker creates new files everywhere it's executed and I don't want them to be all over the repository tree. 

I put it in a list for consistency reasons. As far as I can tell most of the other properties take lists as arguments, when defining a syntax checker. We could allow both as well if you'd like. (string and string in list)